src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -245,12 +245,7 @@ impl<'a> Dependency for (&'a PackageId, Stage) {
         let (id, stage) = *self;
         let pkg = packages.iter().find(|p| p.package_id() == id).unwrap();
         let deps = resolve.deps(id).into_iter().flat_map(|a| a)
-                          .filter(|dep| *dep != id)
-                          .map(|dep| {
-                              (dep, pkg.dependencies().iter().find(|d| {
-                                  d.name() == dep.name()
-                              }).unwrap())
-                          });
```

Comment:
Could you generate a patch that I could apply locally? I may have to poke a bit as well